### PR TITLE
adding translations

### PR DESCRIPTION
**Pull request by emre2038**
## Description
Translating texts on the fly using `i18next` library. Assumed data structure is as follows :
```
            this.translations = {
                en: {
                    translation: {
                        'Point Mapper': 'Service Mapper',
                        'lorem ipsum': 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
                                   Nullam a lacinia diam. Aliquam sollicitudin lectus ipsum,'
                    }
                }
            }
```

## Related Issue
https://github.com/OpenUpSA/wazimap-ng-ui/issues/237

## Screenshots
before translations : 
image

after translations : 
image

### Added
* `setup/translations.js` to handle translation related events - for now it fires `translate()` function on `profile.loaded` event
* `elements/translations.js` to translate the html elements that have `.i18n` class 

### Updated
* `package.json` to add `i18next` library to the project
* `index.html` to modify the elements that should be translated
* `load.js` to add references of the new classes

## Checklist

- [x]  🚀 is the code ready to be merged and go live?
- [x]  🛠 does it work (build) locally
- [x] 👩‍🎨 does the design matches the [Demo](https://wazimap-ng-v1.webflow.io/demo)

### Pull Request

- [x]  📰 good title
- [x]  📝good description
- [x]  🔖 issue linked
- [x]  📖 changelog filled out
- [x] commit messages are meaningful

### Code Quality

- [x]  🚧 no commented out code
- [x]  🖨 no unnecessary logging
- [x]  🎱 no magic numbers

### Testing

- [ ]  ✅ added (appropriate) unit tests
- [ ]  💢 edge cases in tests were considered
- [ ]  ✅ ran tests locally & are passing

### CODE REVIEW

File: src/js/elements/translations.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+export class Translations extends Observable {
+    constructor(translations) {
+        super();
+
+        this.translations = translations;
+    }
+
+    translate() {
+        //todo:remove this when the BE is ready
```
Comment (milafrerichs) [inline src/js/elements/translations.js:12]: can you already remove this? BE is kind off ready.

**Review by milafrerichs — APPROVED** [review]

File: src/index.html [hunk 2]
```diff
@@ -735,11 +735,11 @@ <h3 class="facility-info__title">Loading...</h3>
               </svg></div>
           </div>
           <div class="point-data__header_title">
-            <div>Point Mapper </div>
+            <div class="i18n" data-i18n="Point Mapper">Point Mapper </div>
           </div>
         </div>
         <div class="point-mapper-content__description">
-          <div>Select the category, or specific type of point data you would like to overlay onto the map.<strong><br></strong></div>
+          <div class="i18n" data-i18n="lorem ipsum">Select the category, or specific type of point data you would like to overlay onto the map.<strong><br></strong></div>
```
Comment (milafrerichs) [inline src/index.html:742]: can you remove the `lore ipsum`  it will fallback to it when it's not defined.

File: src/js/elements/translations.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import {Observable} from "../utils";
+import i18next from 'i18next';
+
+export class Translations extends Observable {
+    constructor(translations) {
+        super();
+
+        this.translations = translations;
```
Comment (milafrerichs) [inline src/js/elements/translations.js:8]: can we make this more easy on the user.
currently I would need to define the following:
```json
"translations": {
    "en": {
      "translation": {
        "Point Mapper": "Services"
      }
    }
  },
```
can we turn this into something more like this: (removing the translation key)

```json
"translations": {
    "en": {
      "Point Mapper": "Services"
    }
  },
```

you would then need to change this line here to something like this:
```js
this.translations = {};
Object.keys(translations).forEach((key) => {
  this.translations[key] = { translations: this.translations[key] }
})
```
or a better solution. this is just the fastest I came up with :)

**Review by milafrerichs — CHANGES_REQUESTED** [review]

**Review by milafrerichs — APPROVED** [review]
